documentation/source/ObjectDetection.md

```diff
@@ -0,0 +1,312 @@
+In order to use `DetectionMetrics` you have to pass a so-called `post_prediction_callback` to the metric, which is responsible for the postprocessing of the model's raw output into final predictions and is explained below. 
+
+### Postprocessing
+
+Postprocessing refers to a process of transforming the model's raw output into final predictions. Postprocessing is also model-specific and depends on the model's output format.
+For `YOLOX` model, the postprocessing step is implemented in [YoloPostPredictionCallback](https://docs.deci.ai/super-gradients/docstring/training/models/#training.models.detection_models.yolo_base.YoloPostPredictionCallback) class.
+It can be passed into a `DetectionMetrics` as a `post_prediction_callback`. 
+The postprocessing of all detection models involves non-maximum suppression (NMS) which filters dense model's predictions and leaves only boxes with the highest confidence and suppresses boxes with very high overlap 
+based on the assumption that they likely belong to the same object. Thus, a confidence threshold and an IoU threshold must be passed into the postprocessing object.
+
+```python
+from super_gradients.training.models.detection_models.yolo_base import YoloPostPredictionCallback
+
+
+post_prediction_callback = YoloPostPredictionCallback(conf=0.001, iou=0.6)
+```
+
+### Visualization
+
+Visualization of the model predictions is a very important part of the training process for any computer vision task. 
+By visualizing the predicted boxes, developers and researchers can identify errors or inaccuracies in the model's output and adjust the model's architecture or training data accordingly.
+
+SuperGradients provide an implementation of [DetectionVisualizationCallback](https://docs.deci.ai/super-gradients/docstring/training/utils/#training.utils.callbacks.callbacks.DetectionVisualizationCallback). 
+You can use this callback in your training pipeline to visualize predictions during training. For this, just add it to `training_hyperparams.phase_callbacks` in your yaml.
+During training, the callback will generate a visualization of the model predictions and save it to the TensorBoard or Weights & Biases depending on which logger you
+are using (Default is Tensorboard). 
+
+If you would like to do the visualization outside of training you can use `DetectionVisualization` class as follows:
+```python
+import torch
+import numpy as np
+
+from super_gradients.training import models
+from super_gradients.training.utils.detection_utils import DetectionVisualization
+from super_gradients.training.datasets.datasets_conf import COCO_DETECTION_CLASSES_LIST
+
+
+def my_undo_image_preprocessing(im_tensor: torch.Tensor) -> np.ndarray:
+    im_np = im_tensor.cpu().numpy()
+    im_np = im_np[:, ::-1, :, :].transpose(0, 2, 3, 1)
+    im_np *= 255.0
+    return np.ascontiguousarray(im_np, dtype=np.uint8)
+
+
+model = models.get("yolox_s", pretrained_weights="coco", num_classes=80)
+imgs, targets = iter(train_dataloader).__next__()
```

Comment:
Is there a difference between `next(iter(train_dataloader))` and `__next__()`?
If there is none, then I suggest to use the `next` variant as it's more common.

Reply:
Good points, looks much better, thanks 6e116f9